shared_model/interfaces/common_objects/pager.hpp

```diff
@@ -62,13 +62,19 @@ namespace shared_model {
         return oldStylePager;
       }
 
-      template <typename T, typename OldModelQueryType>
-      DEPRECATED static void initAllocatedOldModel(
-          const T &obj, OldModelQueryType *oldModel) {
+      /**
+       * Assign new model pager to old model pager without new allocation
+       * @tparam OldModel - Old model pager type
+       * @tparam Model - New model pager type
+       * @param oldModelPager - pointer type of old model pager
+       * @param newModelPager - type of new model pager (interface)
+       */
+      template <typename OldModel, typename Model>
+      DEPRECATED static void setAllocatedPager(OldModel *oldModelPager,
```

Comment:
First, we need to wait until [this](https://github.com/hyperledger/iroha/pull/902) will be merged into `develop`. 
After it, wrap pager with ifndef for old-model. This requires because external APIs, such as Java, JS, Python shouldn't depend on Iroha's internals.

Reply:
Done.
About rebasing onto develop, it needs too much rebasing (even if trunk branch is rebased this year) so I think it's better at last.